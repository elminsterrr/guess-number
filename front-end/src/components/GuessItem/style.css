.GuessItem-span {
  color: #dc143c;
}
