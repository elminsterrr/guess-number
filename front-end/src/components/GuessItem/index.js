import React, { Component } from 'react';
import './style.css';

class GuessItem extends Component {
  render() {
    const { newNumber, resultCode } = this.props;

    if (resultCode === 'success') {
      return <div />;
    }

    return (
      <div>
        Guess is <span className="GuessItem-span">{newNumber}</span> and we must
        search for the <span className="GuessItem-span">{resultCode}</span>{' '}
        number!
      </div>
    );
  }
}

export default GuessItem;
