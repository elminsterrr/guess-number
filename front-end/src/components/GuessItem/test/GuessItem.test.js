import React from 'react';
import { shallow } from 'enzyme';
import GuessItem from '../';

describe('GuessItem', () => {
  const guessItem = shallow(<GuessItem />);

  it('renders correctly', () => {
    expect(guessItem).toMatchSnapshot();
  });
});
