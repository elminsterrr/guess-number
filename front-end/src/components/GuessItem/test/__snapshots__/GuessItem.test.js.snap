// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GuessItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Guess is ",
        <span
          className="GuessItem-span"
/>,
        " and we must search for the ",
        <span
          className="GuessItem-span"
/>,
        " ",
        "number!",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Guess is ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "GuessItem-span",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      " and we must search for the ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "GuessItem-span",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      " ",
      "number!",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Guess is ",
          <span
            className="GuessItem-span"
/>,
          " and we must search for the ",
          <span
            className="GuessItem-span"
/>,
          " ",
          "number!",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Guess is ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "GuessItem-span",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        " and we must search for the ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "GuessItem-span",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        " ",
        "number!",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
